File: notifier/src/main/java/com/juanimoli/strictmodenotifier/detector/CleartextNetworkDetector.java
package com.juanimoli.strictmodenotifier.detector;

import com.juanimoli.strictmodenotifier.StrictModeLog;

public class CleartextNetworkDetector implements Detector {
    @Override
    public boolean detect(StrictModeLog log) {
        return log.message.contains("CLEARTEXT communication not supported:");
    }
}

File: notifier/src/main/java/io/github/juanimoli/strictmode/notifier/detector/CleartextNetworkDetector.kt
package io.github.juanimoli.strictmode.notifier.detector

import io.github.juanimoli.strictmode.notifier.StrictModeLog

class CleartextNetworkDetector : Detector {
    private val detectMessageList = listOf(
        "Cleartext HTTP traffic to (.*) not permitted".toRegex(),
        "CLEARTEXT communication not supported:".toRegex()
    )

    override fun detect(log: StrictModeLog): Boolean {
        return detectMessageList.any { it.find(log.message) != null }
    }
}
